js/main.js

```diff
@@ -163,12 +187,35 @@ $nextButton.addEventListener('click', function (event) {
   $pageNumberBot.textContent = pageNumber;
   $backButton.hidden = false;
   window.scrollTo({ top: 0, behavior: 'smooth' });
+  getData(pageUrl);
 
   if (pageUrl === null) {
     $nextButton.hidden = true;
   }
+});
 
-  getData(pageUrl);
+$backLink.addEventListener('click', function (event) {
+  view = 'featured';
+  $featuredView.hidden = false;
+  $detailView.hidden = true;
+});
+
+$topLink.addEventListener('click', function (event) {
+  window.scrollTo({ top: 0, behavior: 'smooth' });
+});
+
+$searchIcon.addEventListener('click', toggleModal);
+$closeButton.addEventListener('click', toggleModal);
+
+$searchInput.addEventListener('keyup', function (event) {
+  var searchUrl = domain + key + searchParam + $searchInput.value;
+  $resultsView.hidden = false;
+
+  if (event && timeoutId !== null) {
+    clearTimeout(timeoutId);
+  } else {
+    timeoutId = setTimeout(getData(searchUrl), 500);
```

Comment:
Uh oh, this is calling `getData` immediately, it's not actually waiting to send the request. You can fix this by changing the first argument to `function(){ getData(searchUrl) }`